### PR TITLE
Implemented XML export of transponder and channel list

### PR DESCRIPTION
**Pull request by jmwislez**
Full implementation of the XML export as per current transponder struct content and XML DTD. 
Tested on DVB-S/S2 (Astra 19.2E, Astra 23.5E, Astra 28.2E, Hotbird 13.0E, Eutelsat 5.0W) and XML validates.

### CODE REVIEW

File: src/dump-xml.c [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*
  * Simple MPEG/DVB parser to achieve network/service information without initial tuning data
  *
- * Copyright (C) 2006-2014 Winfried Koehler
+ * Copyright (C) 2006-2014 Winfried Koehler, 2021 Jean-Marc Wislez
```
Comment (stefantalpalaru) [inline src/dump-xml.c]: Put your copyright info on a new line, like this:

```text
* Copyright (C) 2021 Jean-Marc Wislez
```

**Comment by stefantalpalaru** [comment]
We need consistent indentation in our source files. A while ago, I settled on the Linux kernel rules for all the code: https://www.kernel.org/doc/html/v4.10/process/coding-style.html

While you figure out how to get your text editor to enforce those for you, feel free to run this script on the files you modify: https://github.com/torvalds/linux/blob/master/scripts/Lindent

Also, try looking into what changes the permissions of all the files you edit to 755.

**Comment by jmwislez** [comment]
I believe that with these commits, the points you made are now addressed.  Any further concerns?

**Comment by stefantalpalaru** [comment]
> Any further concerns?

No, I just need to find the time to review it.

**Comment by jmwislez** [comment]
No prob, take your time!

**Comment by stefantalpalaru** [comment]
Merged. Thanks!

File: src/dump-xml.c [hunk 2]
```diff
@@ -35,68 +42,434 @@ typedef struct {
...
+	else if (id == 0x4AE0)
+		return "DRE-Crypt";
+	else if (id == 0x4AE1)
+		return "DRE-Crypt";
+	else if (id == 0x4AE4)
+		return "CoreCrypt";
+	else if (id == 0x4AEA)
+		return "Cryptoguard";
+	else if (id == 0x4AEB)
+		return "Abel Quintic";
```
Comment (Saentist) [inline src/dump-xml.c:180]: else if (id == 0x4AEE)
		return "Bulcrypt";

Comment (stefantalpalaru) [reply src/dump-xml.c:180]: Can you make a new PR?

Comment (Saentist) [reply src/dump-xml.c:180]: > Can you make a new PR?

Yes after merge this one. 
There is also missing 
CAID 1010
and Verimatrix variants
https://trac.streamboard.tv/oscam/wiki/CardsList
old resource not updated for long time

----
But there is something missing to be complete 
CAID ID is OK but Provider ID is missing
aka:
`4AEE:000000
`and
`4AEE:000001
`

**Pull request merged**